PixelHunter1995/Inputs/SignalState.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PixelHunter1995.Inputs
+{
+
+    // TODO consider class vs struct. Overridden methods makes class work in dict anyway... while struct is value-type.
+    // TODO As Input creates an "alias" for it thorugh inheritance, it needs to be class for now.
+    /// <summary>
+    /// The state of a signal-graph, intended to use for keypressed as monogame does not give any information on
+    /// whether the key was released or pressed.
+    /// Rather than an enum like one would expect, this is a class, meaning each ProperKeyState is an instance.
+    /// There are static references for each permutation to instances describing them, to allow easier use.
+    /// 
+    /// The advantage with doing it this way however, is that class-properties can be used to nicely get a bool
+    /// for various properties of the keystate.
+    /// So `this.GetKeyState(key) == SignalState.EdgeDown` can be written as `this.GetKeyState(key).IsEdgeDown`.
+    /// As a result however, it makes much less sense to enable 'Is' syntax (ie `input.IsKeyState(key, state)`),
+    /// although nothing stops one from implementing that as options/aliases of the former.
+    /// 
+    /// The disadvantage could have been that one has to be careful about making sure to use the static references,
+    /// as otherwise `==` or `.Equals` would compare reference and not value.
+    /// This has however been rendered moot by overriding the relevant methods,
+    /// and could also have been solved by turning it into a struct (making it a value-type).
+    /// </summary>
+    class SignalState
+    {
+        public bool IsUp { get; }
+        public bool IsEdge { get; }
+        public bool IsDown { get => !this.IsUp; }
+        public bool IsHeld { get => !this.IsEdge; }
```

Comment:
I would say that a Held key is one that is held down.

Reply:
I needed an antonym to "Edge" in the context of "IsEdge" vs "Is*", and "Held" is the best I could think of.
This way it allows to check for "IsHeldUp" and "IsHeldDown".
Feel free to give suggestion to another word that fits. I were thinking IsFlat for a while, but felt it sounded off somehow.